pkg/specgen/specgen.go

```diff
@@ -191,6 +191,10 @@ type ContainerBasicConfig struct {
 	// The execution domain system allows Linux to provide limited support
 	// for binaries compiled under other UNIX-like operating systems.
 	Personality *spec.LinuxPersonality `json:"personality,omitempty"`
+	// IsInfra denotes that we are creating the infra container
+	IsInfra bool `json:"is_infra,omitempty"`
```

Comment:
I think this is necessary (at least for now) because certain aspects of specgen should and should not be applied to an infra container spec. If you look at the references of `IsInfra` I use it frequently in container validation and namespace settings because a lot of the namespace settings depend on `pod != nil` but if these code paths execute, they will try to grab infra, which won't work because infra is not created yet.

Reply:
Why is this validation happening in Specgen and not inside Libpod?

Reply:
once we get inside of libpod, the spec has already picked up some crucial things that infra (at the moment) should not have. Once I go through and implement all other pod create features, I will most likely be able to pull this out. The biggest issue right now is bind mounts (which are next on my list) and namespace options.

Reply:
You should be able to move this validation into `container_validate.go`

Reply:
Yeah, I think we can but that would make `container_validate` really complicated and it would end up undoing a lot of the work `container_create` did. if you look at `MakeContainer()` you will see that the reason it is in the spec is so we avoid making mistakes that would be correct with a normal container. `IsInfra` is basically a mechanism to bar certain codepaths from executing at the moment.

Reply:
keeping things on the specgen side rather than internal libpod seems to agree more with the container creation process which is what I am trying to mimic. We can discuss more next week, I am open to changing it but I think it kind of doubles the amount of work the underlying code does

Reply:
okay, got it down to the point where `s.IsInfra` is really only used to 
1) make sure that pod == nil in `MakeContainer`
 2) leave out the container create command default 
(os.args) 
3) call `withIsInfra` from within libpod once the other container create options are established.

Reply:
The reason I am pushing so hard here is that Specgen is part of our public API. We cannot add and remove elements as we please without breaking changes (and resultant major version bumps), and this absolutely should not be part of the public API - the creation of infra containers is purely an internal Libpod matter, users should not be able to specify it themselves.

Reply:
Ah I see, that makes a lot of sense thanks for the explanation. I was able to get rid of all `infra` related entities in specgen. I did this by splitting `MakeContainer` off at the end so I can modify the infra specgen and options back in `pod_create`. I then call a new function called `ExecuteCreate` which does what the end of `MakeContainer` used to do, now with the ability to have a proper infra config.